src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,698 @@
+        rs = 60 * (1500 + ZonePPFD) / (200 + ZonePPFD);
+        ra = 350 * std::pow((0.1 / 0.1), 0.5) * (1 / (LAI + 1e-10));
+        ETRate =
+            (1 / hfg) * (slopepat * (In - G) + (rhoair * CpAir * ZoneVPD) / ra) / (slopepat + psyconst * (1 + rs / ra)); // Penman-Monteith ET model
+        return ETRate;                                                                                                   // mm/s; kg/(m2s)
+    }
+    Real64 ETStanghellini(EnergyPlusData &state, Real64 ZonePreTemp, Real64 ZonePreHum, Real64 ZoneCO2, Real64 ZonePPFD, Real64 ZoneVPD, Real64 LAI)
+    {
+        // PURPOSE OF THIS SUBROUTINE:
+        // This subroutine is for using Stanghellini ET model to calculate evapotranspiration rates from the Indoor Greenery System objects.
+        // Reference: Stanghellini, C., Transpiration of greenhouse crops: an aid to climate management, 1987, Institute of Agricultural Engineering,
+        // Wageningen, The Netherlands
+
+        // SUBROUTINE PARAMETER DEFINITIONS:
+        static constexpr std::string_view RoutineName("ETStanghellini: ");
+        Real64 hfg = Psychrometrics::PsyHfgAirFnWTdb(ZonePreHum, ZonePreTemp) / std::pow(10, 6); // Latent heat of vaporization (MJ/kg)
+        Real64 slopepat =
+            0.200 * std::pow((0.00738 * ZonePreTemp + 0.8072), 7) - 0.000116; // Slope of the saturation vapor pressure-temperature curve (kPa/°C)
+        Real64 CpAir = Psychrometrics::PsyCpAirFnW(ZonePreHum) / std::pow(10, 6); // specific heat of air at constant pressure (MJ kg−1 °C−1)
+        Real64 OutPb = state.dataEnvrn->OutBaroPress / 1000;                      // outdoor pressure (kPa)
+        Real64 constexpr mw(0.622);                                               // ratio molecular weight of water vapor / dry air = 0.622.
+        Real64 psyconst = CpAir * OutPb / (hfg * mw);                             // Psychrometric constant (kPa/°C)
+        Real64 rs = 0.0;                                                          // stomatal resistance s/m
+        Real64 ra = 0.0;                                                          // aerodynamic resistance s/m
+        Real64 In = ZonePPFD * 0.327 / std::pow(10, 6);                           // net radiation MW/m2
+        Real64 G = 0.0;                                                           // soil heat flux (MJ/(m2s))
+        Real64 rhoair = Psychrometrics::PsyRhoAirFnPbTdbW(state, OutPb * 1000, ZonePreTemp, ZonePreHum); // kg/m3
+        Real64 ETRate;                                                                                   // mm/s; kg/(m2s)
+        rs = 60 * (1500 + ZonePPFD) / (200 + ZonePPFD);
+        ra = 350 * std::pow((0.1 / 0.1), 0.5) * (1 / LAI);
+        ETRate = (1 / hfg) * (slopepat * (In - G) + (2 * LAI * rhoair * CpAir * ZoneVPD) / ra) /
+                 (slopepat + psyconst * (1 + rs / ra)); // Stanghellini ET model
+        return ETRate;                                  // mm/s; kg/(m2s)
+    }
```

Comment:
These two functions look too similar.  I think you can collapse almost all of the contents into a single function and just have a switch to behave differently.

Reply:
Thank you! Combined into one function.

Reply:
So both functions were identical?